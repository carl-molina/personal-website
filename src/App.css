@import 'normalize.css';

@keyframes slideIn {
  from {
    opacity: 0;
    transform: translateY(-30px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

body {
  background-color: #000300;
}

.App {
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: crimson;
  text-align: center;
}

.App-logo {
  height: 20vmin;
}

.App-link {
  color: darkgray;
}

.tree-bg {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-direction: row;
  text-align: center;
  flex-wrap: wrap;
}

.mountains-fg {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-direction: row;
  text-align: center;
  flex-wrap: wrap;
}

h1, h2, h3, h4, p {
  color: white;
  background-color: #003a47;
  padding: 5px;
  margin: 0.5em 0;
}

img.profile-img {
  width: 33%;
  border-radius: 50%;
  border: 2px double rgb(0, 44, 39);
}

.container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  animation: slideIn 1s ease;
}

.Intro-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  text-align: start;
  margin-left: 20px;
  color: white;
}

.Body-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin-left: 20px;
  color: white;
}

@media (max-width: 768px) {
  .container {
    flex-direction: column;
    align-items: center;
  }

  .Intro-container {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    text-align: center;
    margin-left: 20px;
  }

}

.css-button-arrow--green {
  min-width: 130px;
  height: 40px;
  color: #fff;
  padding: 5px 10px;
  font-weight: bold;
  cursor: pointer;
  transition: all 0.3s ease;
  position: relative;
  display: inline-block;
  outline: none;
  overflow: hidden;
  border-radius: 5px;
  border: none;
  background-color: #003a47;
}
.css-button-arrow--green:hover {
  border-radius: 5px;
  padding-right: 24px;
  padding-left:8px;
}
.css-button-arrow--green:hover:after {
  opacity: 1;
  right: 10px;
}
.css-button-arrow--green:after {
  content: "\00BB";
  position: absolute;
  opacity: 0;
  font-size: 20px;
  line-height: 40px;
  top: 0;
  right: -20px;
  transition: 0.4s;
}

/* .bottom-component {
  z-index: 10;
  width: 100%;
}

.after-parallax {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  min-height: 100vh;
} */


.Parallax-main {
  display: block;
}



/* .bg-image {
  width: 100%;
  min-width: 100vw;
  min-height: 100vh;
  object-fit: cover;
}

.fg-image {
  width: 100%;
  min-width: 100vw;
  min-height: 100vh;
  object-fit: fill;
} */