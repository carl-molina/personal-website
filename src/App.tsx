import React from "react";
import { Parallax, ParallaxLayer } from "@react-spring/parallax";
import { BrowserRouter, Routes, Route } from "react-router-dom";

import bg from "./images/IMG_1458.jpg";
import fg from "./images/layered-peaks-haikei.svg";
import "./App.css";

import Navbar from "./Navbar";
import Intro from "./Intro";
import Body from "./Body";
import Projects from "./Projects";
import Contact from "./Contact";

/** Component for entire page.
 *
 * Props: none
 * State: none
 *
 * App -> { Navbar, Intro, Body }
 */

function App(): React.ReactElement {

  function handleScrollAboutMe() {
    const aboutMe = document.getElementById("about-me");
    aboutMe?.scrollIntoView({ behavior: "smooth" });
  }

  return (
    <>
      <BrowserRouter>
        <main>
          <>
            <div className="keyart">

              <Navbar />
              <Parallax pages={2} className="Parallax-main">
                <ParallaxLayer
                  className="tree-bg"
                  offset={0}
                  speed={0.5}
                  factor={2.5}
                  style={{
                    backgroundImage: `url(${bg})`,
                    backgroundSize: 'cover',
                  }}
                >

                  <div className="container">
                    <img
                      className="profile-img col-8 mt-5"
                      src="./src/images/CarlMolina_ProfilePhoto4.jpg"
                      alt="profile photo"
                    />
                    <div className="Intro-container col-4 mt-5 ml-5 mr-5">
                      <Routes>
                        <Route path="/" element={<Intro />} />
                        <Route path="/projects" element={<Projects />} />
                        <Route path="/contact" element={<Contact />} />
                      </Routes>
                      <button
                        className="css-button-arrow--green"
                        onClick={handleScrollAboutMe}
                      >
                        More About Me
                      </button>
                    </div>
                  </div>
                </ParallaxLayer>
                <ParallaxLayer
                  className="mountains-fg"
                  offset={1}
                  speed={2}
                  factor={1}
                  style={{
                    backgroundImage: `url(${fg})`,
                    backgroundSize: 'cover',
                    backgroundRepeat: "repeat"
                  }}
                >
                  <div className="Body-container" id="about-me">
                    <Body />
                  </div>
                </ParallaxLayer>
              </Parallax>
            </div>
          </>
        </main>
      </BrowserRouter>
    </>
  );
};

export default App;
