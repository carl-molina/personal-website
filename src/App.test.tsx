import React from "react";
import { test, expect, describe } from "vitest";
import { render, screen, fireEvent } from '@testing-library/react';

import App from './App';


describe("App smoke test", function () {
  test('renders without crashing', function () {
      render(<App />);
  });
});


describe("App snapshot test", function () {
  test('matches snapshot', function () {
      const { container } = render(<App />);
      expect(container).toMatchSnapshot();
  });
});
