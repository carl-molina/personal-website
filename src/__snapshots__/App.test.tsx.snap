// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Projects snapshot test > matches snapshot 1`] = `
<div>
  <main>
    <div
      class="keyart"
    >
      <div
        class="Navbar"
      >
        <nav>
          <a
            aria-current="page"
            class="Navbar-title active"
            href="/"
          >
            Carl Molina
          </a>
          <ul>
            <li>
              <a
                download="Carl_Molina_Resume.pdf"
                href="/public/Carl_Molina_Resume.pdf"
              >
                Resume
              </a>
            </li>
            <li>
              <a
                class=""
                href="/projects"
              >
                Projects
              </a>
            </li>
            <li>
              <a
                class=""
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="Parallax-main"
        style="position: absolute; width: 100%; height: 100%; overflow-y: scroll; overflow-x: hidden; transform: translate3d(0px,0px,0px);"
      >
        <div
          style="overflow: hidden; position: absolute; width: 100%; height: 0px; transform: translate3d(0px,0px,0px);"
        >
          <div
            class="tree-bg"
            style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; background-size: cover; background-repeat: no-repeat; will-change: transform; width: 100%; height: 0px; transform: translate3d(0,0px,0); background-image: url(/public/IMG_1458.jpg);"
          >
            <div
              class="container"
            >
              <img
                alt="profile photo"
                class="profile-img col-8 mt-5"
                src="/public/CarlMolina_ProfilePhoto4.jpg"
              />
              <div
                class="Intro-container col-4 mt-5 ml-5 mr-5"
              >
                <div
                  class="RoutesList"
                >
                  <h1>
                    Software Engineer
                  </h1>
                  <h3>
                    Sunnyvale, CA
                  </h3>
                  <p>
                    Former Field Op Specialist with calibration and sensor health experience in the autonomous vehicle industry.
                    <br />
                    <br />
                    Now full-stack software developer building React-TypeScript apps, automating tests, andÂ 
                    <i>
                      <a
                        class="Medium-post"
                        href="https://medium.com/@carlandrew.molina
                    /ordering-a-burritozilla-in-the-middle-of-a-coding-bootcamp
                    -15dadf1eca22"
                        rel="noopener"
                        target="_blank"
                      >
                        deciding whether or not to eat a Burritozilla for the umpteenth time.
                      </a>
                    </i>
                  </p>
                </div>
                <button
                  class="css-button-arrow--green"
                >
                  More About Me
                </button>
              </div>
            </div>
          </div>
          <div
            class="mountains-fg"
            style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; background-size: cover; background-repeat: repeat; will-change: transform; width: 100%; height: 0px; transform: translate3d(0,0px,0); background-image: url(/public/layered-peaks-haikei.svg);"
          >
            <div
              class="Body-container"
              id="about-me"
            >
              <div
                class="Body-container"
              >
                <h2
                  class="Body-header mt-5"
                >
                  Experience
                </h2>
                <h3>
                  Field Operator Specialist III
                </h3>
                <h4
                  class="Kett"
                >
                  <a
                    href="https://www.ketteng.com/"
                    rel="noopener"
                    target="_blank"
                  >
                    Kett Engineering 2018-2023
                  </a>
                </h4>
                <h2
                  class="Body-header mt-5"
                >
                  Education
                </h2>
                <h3
                  class="RS-title mt-4"
                >
                  Full Stack Web Development
                </h3>
                <h4
                  class="RS"
                >
                  <a
                    href="https://www.rithmschool.com/"
                    rel="noopener"
                    target="_blank"
                  >
                    Rithm School 2023-2024
                  </a>
                </h4>
                <h3
                  class="De-Anza mt-4"
                >
                  Associate of Science, Computer Science
                </h3>
                <h4
                  class="RS"
                >
                  <a
                    href="https://www.deanza.edu/"
                    rel="noopener"
                    target="_blank"
                  >
                    De Anza College 2019
                  </a>
                </h4>
                <h3
                  class="LMU mt-4"
                >
                  Bachelor of Arts, Screenwriting
                </h3>
                <h4
                  class="RS"
                >
                  <a
                    href="https://www.lmu.edu/"
                    rel="noopener"
                    target="_blank"
                  >
                    Loyola Marymount University 2016
                  </a>
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
